### PR TITLE
Apply `@AllowedValues` constraint on all search filters

### PR DESCRIPTION
**Pull request by vierbergenlars**
For search filters that have a different name than the property itself,
also create the options for AllowedValues, so a select can be shown for
them as well

### CODE REVIEW

File: contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/hal/forms/ContentGridHalFormsConfigurationTest.java [hunk 1]
```diff
@@ -170,9 +219,7 @@ void updateFormFieldForConstrainedAttributeOptions() throws Exception {
                                             name: "gender",
                                             type: "radio",
                                             options: {
-                                                inline: [ "female", "male" ],
-                                                minItems: 0,
-                                                maxItems: 1
```
Comment (NielsCW) [inline contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/hal/forms/ContentGridHalFormsConfigurationTest.java:175]: Why was this removed?

Comment (rschev) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/hal/forms/ContentGridHalFormsConfigurationTest.java:175]: The json payload for this test still contains minItems: 0, maxItems: 1, it's now just only tested in the search requests.

Comment (vierbergenlars) [reply contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/data/rest/hal/forms/ContentGridHalFormsConfigurationTest.java:175]: I think this was a rebase error; I pulled one commit out into 2 PRs. There is no reason why this shouldn't be tested.

**Review by rschev — APPROVED** [review]

**Review by NielsCW — APPROVED** [review]
